test/integration/targets/dms_replication_subnet_group/tasks/main.yml

```diff
@@ -0,0 +1,151 @@
+---
+
+- name: set connection information for all tasks
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      region: "{{ aws_region }}"
+    resource_prefix: "test_dms_sg"
+    dms_sg_identifier: "{{ resource_prefix }}-dms"
+  no_log: yes
+
+- block:
+    - name: Create VPC for use in testing
+      ec2_vpc_net:
+        name: "{{ resource_prefix }}-vpc"
+        cidr_block: 10.22.32.0/23
+        tags:
+          Name: Ansible ec2_instance Testing VPC
+        tenancy: default
+        <<: *aws_connection_info
+      register: testing_vpc
+
+    - name: create subnet1
+      ec2_vpc_subnet:
+        state: present
+        vpc_id:  "{{ testing_vpc.vpc.id }}"
+        cidr: 10.22.32.16/28
+        az: eu-west-1a
+        <<: *aws_connection_info
+      register: subnet1
+
+    - name: create subnet2
+      ec2_vpc_subnet:
+        state: present
+        vpc_id:  "{{ testing_vpc.vpc.id }}"
+        cidr: 10.22.32.32/28
+        az: eu-west-1c
+        <<: *aws_connection_info
+      register: subnet2
+
+    - name: create replication subnet group
+      dms_replication_subnet_group:
+        state: present
```

Comment:
This action requires dms-vpc-role.
https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.APIRole.html
https://forums.aws.amazon.com/thread.jspa?messageID=694553
```
botocore.errorfactory.AccessDeniedFault: An error occurred (AccessDeniedFault) when calling the CreateReplicationSubnetGroup operation: The IAM Role arn:aws:iam::112233445566:ro
le/dms-vpc-role is not configured properly.

fatal: [testhost]: FAILED! => {
    "changed": false,
    "error": {
        "code": "AccessDeniedFault",
        "message": "The IAM Role arn:aws:iam::112233445566:role/dms-vpc-role is not configured properly."
    },
```
Examples of creating a required role in a test can be seen in https://github.com/ansible/ansible/tree/devel/test/integration/targets/aws_codepipeline  or https://github.com/ansible/ansible/tree/devel/test/integration/targets/aws_secret/

Reply:
Hello , 

this should now work, would you mind trying again please ?